backend/crates/eiffelvis_core/src/domain/user_queries.rs

```diff
@@ -148,17 +148,19 @@ impl<I> TrackedQuery<I> {
             if self.event_filters.is_empty() {
                 true
             } else {
+                // Event filters itterator that returns nodes that contain the imput given (id filter has to be an exact match, all others do not.)
+                let data = node.data();
                 self.event_filters.iter().filter(|v| !v.is_empty()).any(|filters| {
                     filters.iter().all(|filter| match &filter.pred {
-                        EventFilter::Type { names: ref name } => name.iter().any(|name| &node.data().meta.event_type == name),
+                        EventFilter::Type { names: ref name } => name.iter().any(|name| data.meta.event_type.to_lowercase().contains(&name.to_lowercase())),
```

Comment:
I meant that to_lowercase is a constant in this loop, node.tdata() has no overhead in itself

Reply:
Do you mean something like `let node_type = data.meta.event_type.to_lowercase();` outside the loop for each attribute that `to_lowercase()` is applied to?

Reply:
yes

Reply:
I have managed to more the event type out of the loop but I'm struggling with the source host and name, any suggestions?

Reply:
Thanks! I see where I was going wrong! :)